taichi/runtime/llvm/runtime.cpp

```diff
@@ -691,10 +697,14 @@ void taichi_assert(Context *context, i32 test, const char *msg) {
 const std::size_t ASSERT_MSG_BUFFER_SIZE = 2048;
 char assert_msg_buffer[ASSERT_MSG_BUFFER_SIZE];
 i32 assert_msg_buffer_lock = 0;
+
 void taichi_assert_format(LLVMRuntime *runtime,
                           i32 test,
                           const char *format,
                           ...) {
+  // This function is CPU-only.
+  // It seems that using std::va_list leads to an llvm::SelectionDAG error.
+  // Also note that CUDA cannot call runtime->host_vsnprintf.
```

Comment:
We may use the same function as `PrintStmt` does?

Reply:
On CUDA there's a special `printf` function that allows you to format and print to console. That's what we use in `PrintStmt` on CUDA. However, there's unfortunately no such function to print to a string buffer...